SchoolManagerSystem.Service/Courses/Implementations/CourseServices.cs

```diff
@@ -0,0 +1,160 @@
+﻿using SchoolManagerSystem.Common.DTOs;
+using SchoolManagerSystem.Repository.UnitOfWork.Interfaces;
+using SchoolManagerSystem.Service.Courses.Interfaces;
+using SchoolManagerSystem.Service.Validations.Interfaces;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace SchoolManagerSystem.Service.Courses.Implementations
+{
+	public class CourseServices : ICourseServices
+	{
+		private readonly IUnitOfWork _unit;
+		private readonly IRequestValidations _validate;
+
+		public CourseServices(IUnitOfWork unit, IRequestValidations validate)
+		{
+			_unit = unit;
+			_validate = validate;
+		}
+
+		public async Task<CourseResponse> AddCourse(CourseRequest request)
+		{
+			if (!_validate.IsCourseNameValid(request.CourseName))
+			{
+				throw new ArgumentException("Course Name Is Invalid");
+			}
+
+			var student = await _unit.Student.FetchStudentAsync(request.LevelId);
+			if (student.Count == 0)
+			{
+				throw new ArgumentNullException($"invalid levelId {request.LevelId} ");
+			}
+
+			var course = await _unit.Course.AddCourse(request, student);
+			if (course == null)
+			{
+				throw new ArgumentNullException();
+			}
+			await _unit.SaveChangesAsync();
+			var level = await _unit.Level.FetchLevelAsync(course.LevelId);
+			if (level == null)
+			{
+				throw new ArgumentNullException($"Invalid levelId {course.LevelId}");
+			}
+			var response = new CourseResponse
+			{
+				Id = course.Id,
+				CourseName = course.CourseName,
+				LevelName = level.LevelName
+			};
+
+			return response;
+		}
+
+		public async Task<string> DeleteCourseAsync(string courseId)
+		{
+			var course = await _unit.Course.GetCoursesAsync(courseId);
+			if (course == null)
+			{
+				throw new ArgumentNullException($"User with {courseId} Not Found ");
+			};
+
+			var result = _unit.Course.DeleteCourseAsync(course);
+			await _unit.SaveChangesAsync();
+			if (!result.IsCompleted)
```

Comment:
do you understand the meaning of SaveChanges..
You don't do any check after calling savechanges..
The deleteCourse method should be void.

Reply:
The deleteCourse method should be void. This has not been resolved.. the method should return void.. the Iscomplete check is unneccesary.